src/test/app/SetJSHook_test.cpp

```diff
@@ -9091,6 +9091,125 @@ class SetJSHook_test : public beast::unit_test::suite
+              17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17,
+            ]
+            var ans = [
+              17, 0, 97, 34, 0, 0, 0, 0, 36, 4, 31, 148, 217, 37, 4, 94, 132, 183, 45, 0, 0,
+              0, 0, 85, 19, 64, 179, 37, 134, 49, 150, 181, 111, 65, 245, 137, 235, 125, 47,
+              217, 76, 13, 125, 184, 14, 75, 44, 103, 167, 120, 42, 214, 194, 176, 119, 80,
+              86, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17,
+              17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 98, 64, 0, 0, 0, 0,
+              164, 121, 148, 129, 20, 55, 223, 68, 7, 231, 170, 7, 241, 213, 201, 145, 242,
+              211, 111, 158, 184, 199, 52, 175, 108,
+            ]
+            var ans2 = [
+              17, 0, 97, 34, 0, 0, 0, 0, 36, 4, 31, 148, 217, 37, 4, 94, 132, 183, 45, 0, 0,
+              0, 0, 84, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17,
+              17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 85, 19, 64, 179,
+              37, 134, 49, 150, 181, 111, 65, 245, 137, 235, 125, 47, 217, 76, 13, 125, 184,
+              14, 75, 44, 103, 167, 120, 42, 214, 194, 176, 119, 80, 98, 64, 0, 0, 0, 0,
+              164, 121, 148, 129, 20, 55, 223, 68, 7, 231, 170, 7, 241, 213, 201, 145, 242,
+              211, 111, 158, 184, 199, 52, 175, 108,
+            ]
+            var INVALID_ARGUMENT = -7
+            var TOO_SMALL = -4
+            var PARSE_ERROR = -18
+            var Hook = (arg) => {
+              ASSERT(sto_emplace(void 0, 1, 1) == INVALID_ARGUMENT)
+              ASSERT(sto_emplace(1, 1, void 0) == INVALID_ARGUMENT)
+              ASSERT(sto_emplace(1, 1, -1) == INVALID_ARGUMENT)
+              ASSERT(sto_emplace(1, 1, 4294967215) == INVALID_ARGUMENT)
+              ASSERT(sto_emplace(1, void 0, 1) == INVALID_ARGUMENT)
+              ASSERT(sto_emplace(1, false, 1) == INVALID_ARGUMENT)
+              ASSERT(sto_emplace([0, 1], [0, 1], 327686) == PARSE_ERROR)
+              {
+                ASSERT(sto_emplace([0], [0, 1], 327686) == TOO_SMALL)
+                ASSERT(sto_emplace([0, 1], [0], 327686) == TOO_SMALL)
+                ASSERT(
+                  sto_emplace([0, 1], new Array(4097).fill(0), 327686) == INVALID_ARGUMENT
+                )
+                ASSERT(
+                  sto_emplace(new Array(16385).fill(0), [0, 1], 327686) == INVALID_ARGUMENT
+                )
+              }
+              let buf = []
+              {
+                buf = sto_emplace(sto, ins, 327686)
+                ASSERT(buf.length === sto.length + ins.length)
+                for (let i = 0; i < ans.length; ++i) ASSERT(ans[i] == buf[i])
+                ins[0] = 84
+                buf = sto_emplace(sto, ins, 327684)
+                ASSERT(buf.length == sto.length + ins.length)
+                for (let i = 0; i < ans2.length; ++i) ASSERT(ans2[i] == buf[i])
+              }
+              {
```

Comment:
In the CHook test, there is a comment for `test front insertion`, but it would be easier to compare if there were similar comments here too.
The same for other parts.

Reply:
I didn't actually go over this test that deeply tbh, as it was one that was already done:

https://github.com/Transia-RnD/hooks-toolkit-ts/blob/jshooks-tests/contracts-js/toolbox/tmp-tests/stoEmplace.ts

I do note that the source typescript had lovely 0x hexadecimal literals, but the `hooks-cli compile-js contracts-js/toolbox/base.ts build` does not seem to maintain them in the output `dist/base.js`. 

I can simply copy them ofc, but it may be worth seeing if we can get the `compile-js` command to do "the right thing" here :)